docassemble/ALToolbox/copy_button.py

```diff
@@ -6,18 +6,39 @@ def copy_button_html (
   text_before:str='',
   label:str='Copy',
   tooltip_inert_text:str='Copy to clipboard',
-  tooltip_copied_text:str='Copied!'
+  tooltip_copied_text:str='Copied!',
+  copy_template_block:bool=False,
+  scroll_class:str='', 
+  style_class:str='',
+  adjust_height:str=''
   )->str:
   '''Return the html for a button that will let a user copy the given text'''
+
   button_str = '<div class="al_copy">\n'
   if ( text_before != '' ):
     button_str += f'<span class="al_copy_description">{word( text_before )}</span>\n'
-  button_str += f'<input readonly class="al_copy_value" type="text" value="{ text_to_copy }">\n'
-  button_str += f'<button class="btn btn-primary al_copy_button" type="button">\n'
+
+  # Add textarea tag if copy_template_block is True, along with docassemble template block class names
+  if copy_template_block:
+    button_str += f'<textarea readonly class=" card card-body{style_class} bg-light pb-1 al_copy_value {scroll_class}" placeholder="inert help text here" {adjust_height}>{ text_to_copy }"</textarea>\n'
```

Comment:
I haven't been following quite closely enough, so I'm not sure if I'm missing context. I think this is missing a space and the placeholder text seems like it's not useful here, so I'll make a suggestion

```suggestion
    button_str += f'<textarea readonly class="card card-body {style_class} bg-light pb-1 al_copy_value {scroll_class}" {adjust_height}>{ text_to_copy }"</textarea>\n'
```

A separate question: What would we expect `adjust_height` to be?

Reply:
I see lower down that `adjust_height` is a function.  I'm not a fan of Hungarian notation, but if we anticipate developers using this function directly I think that some indication that this is a function could be helpful.